group-coordinator/src/main/java/org/apache/kafka/coordinator/group/GroupCoordinatorShard.java

```diff
@@ -755,6 +759,26 @@ public void onNewMetadataImage(MetadataImage newImage, MetadataDelta delta) {
         offsetMetadataManager.onNewMetadataImage(newImage, delta);
     }
 
+    private static OffsetCommitKey convertOffsetCommitKeyV0(
```

Comment:
I think the rename to LegacyOffsetCommit is a good idea. This method and the next one for converting the value still use the V0 naming. I suggest renaming to LegacyOffsetCommit too.